### PR TITLE
#6399 - Date/Time formatting bugfix

### LINKED ISSUE #6399: [BUG] Custom date time format does not work in date_time_format
**Describe the bug**
You cannot enter your own Output format when using the predefined date_time_format action

**To Reproduce**
1. Go to Output Format of the date_time_format component
2. Click on Enter a custom expression
3. Enter HH:mm 
4. Click on Test
5. Observe the error

**Expected behavior**
I would expect the custom expression to be able to contain any valid date format string, not just the predefined ones

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Brave
 - Version [e.g. 22] v1.51.114 

**Additional context**

I would like to use HH:mm, since in my case I only need the time portion, but being able to use any valid date time format string would be great.

### PR DESCRIPTION
**Pull request by GTFalcao**
### <samp>🤖 Generated by Copilot at a43fd79</samp>

This pull request improves the error handling and output format options for several formatting actions in the `@pipedream/formatting` package. It uses the `ConfigurationError` class to throw custom errors for invalid or unsupported inputs, and adds a new ISO-like format for date/time values. It also updates the package version and the lock file accordingly.

### <samp>🤖 Generated by Copilot at a43fd79</samp>

> _`ConfigurationError`_
> _Improves user feedback_
> _A winter bug fix_

## WHY

Closes #6399 

### <samp>🤖 Generated by Copilot at a43fd79</samp>

*  Add `ConfigurationError` class to handle invalid user input or configuration for formatting actions ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-1656308c701080730858f43d707af28f8352a324df24e4f4564789e9aac7a8a9R2), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-1801eb79805592c437524df5c3e8014f248e0b7d7997c235d895a82be0dc61afR2), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-5aa0e3b7c51ffc42bd4c1cf49d09ee6a517dd5ee574b03c5570c819166ffdd7eR2))
*  Wrap output formatting logic in `try-catch` blocks and throw `ConfigurationError` if an error occurs ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-1656308c701080730858f43d707af28f8352a324df24e4f4564789e9aac7a8a9L85-R101), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-1801eb79805592c437524df5c3e8014f248e0b7d7997c235d895a82be0dc61afL29-R38), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-5aa0e3b7c51ffc42bd4c1cf49d09ee6a517dd5ee574b03c5570c819166ffdd7eL32-R42))
*  Add a new option for the output date/time format with label "2006-01-22 23:04" and value "YYYY-MM-DD HH:mm" ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-e16fa92a01dc5305f4293939a00734a81b62a122236c3e3505b29e4c26819023R99-R108))
*  Add a semicolon at the end of the `getDateFromInput` function declaration for code style consistency ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-e16fa92a01dc5305f4293939a00734a81b62a122236c3e3505b29e4c26819023L13-R13))
*  Update the version number of the `@pipedream/formatting` package from `0.0.3` to `0.0.4` ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-846687f3a83f906e5a9e53cc9486c91bb2d51f62c1505b5f1eef38f8379c47dbL3-R3))
*  Add new entries for the `components/caspio`, `components/cloze`, `components/hotjar`, `components/respond_io`, `components/retool`, `components/wati`, and `components/wealthbox` packages in the `pnpm-lock.yaml` file to lock the dependencies and versions ([link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR634-R636), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR790-R792), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1932-R1934), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3583-R3585), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3592-R3594), [link](https://github.com/PipedreamHQ/pipedream/pull/6523/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4923-R4928))

### CODE REVIEW

File: components/formatting/actions/add-subtract-time/add-subtract-time.ts [hunk 1]
```diff
@@ -82,17 +83,21 @@ export default defineAction({
...
-      } time`,
-    );
-    return output;
+      $.export(
+        "$summary",
+        `Successfully ${
+          operation === OPERATION_OPTIONS.SUBTRACT
+            ? "subtracted"
+            : "added"
+        } time`,
```
Comment (dylburger) [inline components/formatting/actions/add-subtract-time/add-subtract-time.ts:96]:
```suggestion
        } time`,
```

Comment (dylburger) [reply components/formatting/actions/add-subtract-time/add-subtract-time.ts:96]: Changed "Successfully" to lowercase and removed unnecessary spaces.

File: components/formatting/actions/date-time-format/date-time-format.ts [hunk 2]
```diff
@@ -26,10 +27,14 @@ export default defineAction({
...
 
-    const { outputFn } = DATE_FORMAT_PARSE_MAP.get(outputFormat);
-    const output = outputFn(dateObj);
+    try {
+      const { outputFn } = DATE_FORMAT_PARSE_MAP.get(outputFormat);
+      const output = outputFn(dateObj);
 
-    $.export("$summary", "Successfully formatted date/time");
-    return output;
+      $.export("$summary", "Successfully formatted date/time");
```
Comment (dylburger) [inline components/formatting/actions/date-time-format/date-time-format.ts:34]:
```suggestion
      $.export("$summary", "Successfully formatted date/time");
```

Comment (dylburger) [reply components/formatting/actions/date-time-format/date-time-format.ts:34]: Changed "Successfully" to lowercase.

File: components/formatting/actions/transform-case/transform-case.ts [hunk 3]
```diff
@@ -29,11 +30,15 @@ export default defineAction({
...
-    const { outputFn } = CASE_OPTIONS.find(({ value }) => value === operation);
+    try {
+      const { outputFn } = CASE_OPTIONS.find(({ value }) => value === operation);
 
-    const result = outputFn(input);
+      const result = outputFn(input);
 
-    $.export("$summary", "Successfully transformed text case");
-    return result;
+      $.export("$summary", "Successfully transformed text case");
```
Comment (dylburger) [inline components/formatting/actions/transform-case/transform-case.ts:38]:
```suggestion
      $.export("$summary", "Successfully transformed text case");
```

Comment (dylburger) [reply components/formatting/actions/transform-case/transform-case.ts:38]: Changed "Successfully" to lowercase.

**Review by luancazarine — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hi I can confirm that the action works for both manual format and selected format
image
image

**Pull request merged**